src/units.cpp

```diff
@@ -410,11 +410,13 @@ size_t Units::unitCount() const
 double Units::scalingFactor(const UnitsPtr &units1, const UnitsPtr &units2)
 {
     if ((units1 != nullptr) && (units2 != nullptr)) {
-        double multiplier = 0.0;
+        if ((units1->unitCount() != 0) && (units2->unitCount() != 0)) {
```

Comment:
No need as such for nested `if` statements. IOW, I would rework the code to read something like:

```
if ((units1 != nullptr) && (units2 != nullptr)
    && (units1->unitCount() != 0) && (units2->unitCount() != 0)) {
...
}
```